### PR TITLE
Pc 1347

### PR DESCRIPTION
**Pull request by matrix-drorgo**

### CODE REVIEW

File: modules/ClinikalAPI/src/ClinikalAPI/Service/ApiBuilder.php [hunk 1]
```diff
@@ -99,6 +100,12 @@ public function getApi()
 
                 return $this->getTemplatesForForm($form_id,$form_field,$service_type,$reason_code);
             },
+            "GET /api/indicator-settings" => function () {
+                //exit php or return 401 if not authorized
+                xdebug_break();
```
Comment (matrix-amiel) [inline modules/ClinikalAPI/src/ClinikalAPI/Service/ApiBuilder.php:105]: remove it

File: modules/ClinikalAPI/src/ClinikalAPI/Service/IndicatorSettingsService.php [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    public function getIndicatorSettings($indicator)
+    {
+        xdebug_break();
+        $indicators=array();
+        $indicators['constant'];
+        $indicators['variant'];
+        $listsTable=$this->container->get(ListsTable::class);
+        $lionicListTable =  $listsTable->getAllList($indicator,'seq','ASC');
+        foreach($lionicListTable as $key=>$val){
+            $searchParam = $val['title'];
```
[the remark this answers is not in the document]
Comment (matrix-drorgo) [reply modules/ClinikalAPI/src/ClinikalAPI/Service/IndicatorSettingsService.php]: case '20564-1'://"Oxygen saturation in Blood":
                case '69000-8'://"Heart rate --sitting":
                case '72514-3'://"Pain severity - 0-10 verbal numeric rating [Score] - Reported":
                case '8310-5'://"Body temperature":
                case '8462-4'://"Diastolic blood pressure":
                case '8480-6'://"Systolic blood pressure":
                case '74774-1'://"Glucose [Mass/volume] in Serum, Plasma or Blood":
                case '9303-9'://"Respiratory rate --resting":
.
.
.
 case '8308-9'://"Body height --standing":
                case '8335-2'://"Body weight Estimated":

File: modules/ClinikalAPI/src/ClinikalAPI/Service/IndicatorSettingsService.php [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+
+use \GenericTools\Model\ListsTable;
+
+
+trait IndicatorSettingsService
+{
+    public function getIndicatorSettings($indicator)
+    {
+        xdebug_break();
```
[the remark this answers is not in the document]
Comment (matrix-drorgo) [reply modules/ClinikalAPI/src/ClinikalAPI/Service/IndicatorSettingsService.php]: removed

File: modules/GenericTools/src/GenericTools/Model/ListsTable.php [hunk 4]
```diff
@@ -179,10 +180,27 @@ public function getSpecificRow($listId, $optionId)
         return (array) $row;
     }
 
-    public function getAllList($listId)
-    {
+    public function getAllList($listId,$orderBy = null,$typeOfOrder = null)
+    {xdebug_break();
```
[the remark this answers is not in the document]
Comment (matrix-drorgo) [reply modules/GenericTools/src/GenericTools/Model/ListsTable.php]: removed

File: modules/ClinikalAPI/src/ClinikalAPI/Service/ApiBuilder.php (hunk 1 above)
Comment (matrix-drorgo) [reply modules/ClinikalAPI/src/ClinikalAPI/Service/ApiBuilder.php]: removed

**Pull request merged**